ui/components/CheckboxActions.tsx

```diff
@@ -0,0 +1,81 @@
+import { Tooltip } from "@material-ui/core";
+import _ from "lodash";
+import * as React from "react";
+import styled from "styled-components";
+import { useToggleSuspend } from "../hooks/flux";
+import Button from "./Button";
+import Flex from "./Flex";
+import Icon, { IconType } from "./Icon";
+import Spacer from "./Spacer";
+
+export const makeObjects = (checked: string[], rows: any[]) => {
+  const objects = [];
+  checked.forEach((uid) => {
+    const row = _.find(rows, (row) => {
+      return uid === row.uid;
+    });
+    if (row)
+      return objects.push({
+        kind: row.kind,
+        name: row.name,
+        namespace: row.namespace,
+        clusterName: row.clusterName,
+      });
+  });
+  return objects;
+};
+
+type Props = {
+  className?: string;
+  checked?: string[];
+  rows?: any[];
+};
+
+function CheckboxActions({ className, checked = [], rows = [] }: Props) {
+  //TODO add multi sync
+  const [suspendReqs, setSuspendReqs] = React.useState([]);
+
+  React.useEffect(() => {
+    if (checked.length && rows.length)
+      setSuspendReqs(makeObjects(checked, rows));
+  }, [checked, rows]);
+
+  const resume = useToggleSuspend(
```

Comment:
Just a suggestion (it does not affect the functionality): In order not to repeat the code, which creates button handlers, twice, it is possible to use smth. like:

```
function createSuspendHandler(suspend: boolean) {
    const result = useToggleSuspend(
      {
        objects: suspendReqs,
        suspend: suspend,
      },
      suspendReqs[0] ? suspendReqs[0].kind : ""
    );

    return () => result.mutateAsync();
  }
```

and pass it to buttons like

```
<Button disabled={enabled} onClick={createSuspendHandler(true)}>
```

true for suspend, false for resume

Reply:
Nice - something like this was my initial idea but I couldn't get it all the way there. Returning the function was the piece I was missing!